src/main.cpp

```diff
@@ -794,7 +796,7 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
 
     // Rather not work on nonstandard transactions (unless -testnet/-regtest)
     string reason;
-    if (fRequireStandard && !IsStandardTx(tx, reason))
+    if (!policy.ApproveTx(tx, reason))
```

Comment:
Yes, this is a work in progress. I'm not sure what you have in mind for ApproveTx but I would certainly like to include more things in ApproveTxInputs. In any case, one step at a time.

I don't think we need to mention that the API is not final because it may never be final. Other work in progress (for example, libconsensus) doesn't clarify "the API is not final". In any case, if more people think we should clarify that, I'm happy to include more things in the documentation.